src/pybind/mgr/dashboard/frontend/e2e/cluster/silences.po.ts

```diff
@@ -0,0 +1,100 @@
+import { $, by, element } from 'protractor';
+import { PageHelper } from '../page-helper.po';
+
+export class SilencesPageHelper extends PageHelper {
+  pages = {
+    index: '/#/silence',
+    create: '/#/silence/create'
+  };
+
+  async create(creator, comment, matcher_name, matcher_value) {
+    await this.navigateTo('create');
+
+    await this.waitTextToBePresent(this.getBreadcrumb(), 'Create');
+
+    // enter silence creator
+    await element(by.id('created-by')).clear();
+    await element(by.id('created-by')).sendKeys(creator);
+
+    // enter silence comment
+    await element(by.id('comment')).clear();
+    await element(by.id('comment')).sendKeys(comment);
+
+    const addMatcherButton = element(by.cssContainingText('button', 'Add matcher'));
+    await addMatcherButton.click();
+    // select matcher name
+    const nameField = element(by.id('name'));
+    await nameField.click();
+    await expect(nameField.all(by.css('option')).getText()).toMatch(
+      matcher_name,
+      'Invalid matcher name'
+    );
+    await nameField.element(by.cssContainingText('option', matcher_name)).click();
+
+    // enter matcher value
+    const valueField = element(by.id('value'));
+    await valueField.click();
+    await valueField.sendKeys(matcher_value);
+
+    const addButton = element.all(by.cssContainingText('button', 'Add')).last();
+    await addButton.click();
+    const createButton = element(by.cssContainingText('button', 'Create Silence'));
+    // wait for modal container where you select Matcher to dissapear
+    await this.waitStaleness(element(by.cssContainingText('.modal-dialog', 'Matcher')));
+    await createButton.click();
+    // wait for browser to return to silences page and table to be visible
+    await this.waitPresence(this.getDataTables().first(), 'Timed out waiting for silence creation');
+    // Order silences so most recently updated one is on top. Requires clicking 'Updated' twice
+    await element(
+      by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated')
+    ).click();
+    await element(
+      by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated')
+    ).click();
+    // Opens new silence's details table by clicking on silence then check "comment"
+    // row of details table for silence for the correct comment to verify its existence
+    await this.getFirstTableCellWithText(creator).click();
+    await expect(element(by.cssContainingText('.datatable-body-row', 'comment')).getText()).toMatch(
+      comment
+    );
+  }
+
+  // expire is currently limited to expiring the most recently updated silence with the given creator name
```

Comment:
Usually documentation of classes and methods are documented according to [JSDoc specification](https://devdocs.io/jsdoc/howto-es2015-classes):

```javascript
/**
 * Description of class
 */
class X {
  /**
   * Description of method
   */
  method() {
  }
}
```

The specification contains the recommendation to use `@param` or `@return` keywords. We discourage to use those keywords as a complete documentation can be created out of properly typed TypeScript:

```typescript
class X {
  /*
   * Description
   */
  async expire(creator: string): void {
}
```